retworkx/__init__.py

```diff
@@ -398,6 +398,74 @@ def _graph_dijkstra_shortest_path(
     )
 
 
+@functools.singledispatch
+def all_pairs_dijkstra_shortest_paths(graph, edge_cost_fn):
+    """Find the shortest path from all nodes
+
+    This function will generate the shortest path from all nodes int the graph
+    using Dijkstra's algorithm. This function is multithreaded and will run
+    launch a thread pool with threads equal to the number of CPUs by default.
+    You can tune the number of threads with the ``RAYON_NUM_THREADS``
+    environment variable. For example, setting ``RAYON_NUM_THREADS=4`` would
+    limit the thread pool to 4 threads.
+
+    :param graph: The input graph to use. Can either be a
+        :class:`~retworkx.PyGraph` or :class:`~retworkx.PyDiGraph`
+    :param edge_cost_fn: An optional weight function for an edge. It will
+        accept a single argument, the edge's weight object and will return a
+        float which will be used to represent the weight/cost of the edge
+
+    :return: Dictionary of paths. The keys are destination node indices and
+        the dict values are lists of node indices making the path.
```

Comment:
Would it be important to point out that is a read-only dict?

Reply:
Yep, we should definitely call that out I'll do this now.

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/278/commits/e9ee3657eab1cc02bcfa7cb5829e32ce2a23d1e3